common/djangoapps/third_party_auth/samlproviderconfig/views.py

```diff
@@ -0,0 +1,71 @@
+"""
+Viewset for auth/saml/v0/samlproviderconfig
+"""
+
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from edx_rbac.mixins import PermissionRequiredMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from enterprise.models import EnterpriseCustomerIdentityProvider
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from ..models import SAMLProviderConfig
+from .serializers import SAMLProviderConfigSerializer
+
+
+class SAMLProviderMixin(object):
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = SAMLProviderConfigSerializer
+
+
+class SAMLProviderConfigViewSet(PermissionRequiredMixin, SAMLProviderMixin, viewsets.ModelViewSet):
+    """
+    A View to handle SAMLProviderConfig CRUD
+
+    Usage:
+        [HttpVerb] /auth/saml/v0/provider_config/?enterprise-id=uuid
```

Comment:
Might be worth updating this to be more explicit, `POST /auth/saml/v0/provider_config/?enterprise-id=uuid` won't work for example.

Reply:
Also probably update to match the query string we're expecting in code if this is going to be included